src/go/rdctl/pkg/factoryreset/factory_reset_windows.go

```diff
@@ -22,18 +22,129 @@ import (
 	"os"
 	"os/exec"
 	"path"
+	"path/filepath"
+	"sort"
 	"strings"
 	"syscall"
+	"unsafe"
 
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
 	"golang.org/x/text/encoding/unicode"
 )
 
-func KillRancherDesktop() {
-	cmd := exec.Command("taskkill", "/IM", "Rancher Desktop.exe", "/T", "/F")
-	cmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: CREATE_NO_WINDOW}
-	cmd.Run()
+var (
+	pKernel32      = windows.NewLazySystemDLL("kernel32.dll")
+	pEnumProcesses = pKernel32.NewProc("K32EnumProcesses")
+)
+
+func KillRancherDesktop() error {
+	appDir, err := directories.GetApplicationDirectory()
+	if err != nil {
+		return fmt.Errorf("could not find application directory: %w", err)
+	}
+
+	var processes []uint32
+	err = directories.InvokeWin32WithBuffer(func(size int) error {
+		processes = make([]uint32, size)
+		var bytesReturned uint32
+		// We can't use `windows.EnumProcesses`, because it passes in an incorrect
+		// value for the second argument (`cb`).
+		n, _, err := pEnumProcesses.Call(
+			uintptr(unsafe.Pointer(&processes[0])),
+			uintptr(len(processes)*int(unsafe.Sizeof(processes[0]))),
+			uintptr(unsafe.Pointer(&bytesReturned)),
+		)
+		if n == 0 {
+			return err
+		}
+		processesFound := bytesReturned / uint32(unsafe.Sizeof(uint32(0)))
+		if processesFound >= uint32(size) {
+			return windows.ERROR_INSUFFICIENT_BUFFER
+		}
+		logrus.Tracef("got %d processes", processesFound)
+		processes = processes[:processesFound]
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("could not get process list: %w", err)
+	}
+
+	sort.Slice(processes, func(i, j int) bool {
```

Comment:
Why sort the processes? I don't see a `break` stmt when the loop hits a certain `pid`.

Reply:
It was just there for debugging — having a sorted list makes it easier to spot if a desired one was flagged in the log.